### DISCUSSION TITLE
Do the RDKit indices for a mol generated from an InChI always match the canonical indices of the InChI?

### DISCUSSION THREAD

**Discussion by a-lapsley** [rdkit/rdkit#6830] [Q&A]
If I use the Chem.MolFromInChI() function to generate a Mol object from an InChI, when RDKit generates the Mol object does it do so in a way that means that the indices of the atoms in RDKit match the canonical indices from the InChI? (This is bearing in mind that RDKit starts the indices at 0, while InChI starts at 1, so InChI atom 1 would be RDKit atom 0).

I have done a test on around 1500 molecules and from what I can see the answer seems to be yes, but I wanted to check that this indeed the case.  I have tried looking through the source code myself but unfortunately am not fluent in C++ so was wondering if anyone more knowledgeable about the internal workings of RDKit could help me?

**Comment by dehaenw** [comment]
I think the canonization algorithms are NOT necessarily the same, as inchi is supposed to be universal so should conform to whatever the inchi foundation decides is correct. Also, as an aside, be careful with converting InChI back to Mols, this can lead to problems, see here for an example: https://gist.github.com/dehaenw/3594497d981887f10dcb0220742304d8 and see this blogpost for a more elaborate explanation of some of the problems you might encounter: https://depth-first.com/articles/2021/03/10/parsing-inchi-its-complicated/

**Reply by a-lapsley** [reply]
Is there any canonization that goes on when converting an InChI to a Mol in RDKit? I know RDKit has it's own canonization algorithm built in, which you can call separately (and does indeed give different indices to the InChI one), but I wasn't aware this was used when getting a Mol from an InChI? In the same way that the RDKit indices of a Mol generated from a Molfile are not canonical and depend on the indices in the Molfile.

**Reply by dehaenw** [reply]
I think typically canonization happens only during conversion back to SMILES. I assume when reading inchi to mol it will use the same atom ordering, unless you explicitly canonize somewhere. However, that ordering will not be "rdkit canonical" then.

I think I get what you mean now - i misunderstood you earlier. Yes, I would expect the ordering should be the same.
